pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_NonParam.cpp

```diff
@@ -661,6 +665,45 @@ TEMPLATE_TEST_CASE("StateVectorKokkos::applyToffoli",
     }
 }
 
+TEMPLATE_TEST_CASE("StateVectorKokkos::applyCSWAP",
+                   "[StateVectorKokkos_Nonparam]", float, double) {
+    {
+        using ComplexT = StateVectorKokkos<TestType>::ComplexT;
+        const std::size_t num_qubits = 3;
+
+        StateVectorKokkos<TestType> kokkos_sv{num_qubits};
+
+        kokkos_sv.applyOperations({{"Hadamard"}, {"PauliX"}}, {{0}, {1}},
+                                  {{false}, {false}});
+
+        auto ini_sv = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{},
+                                                          kokkos_sv.getView());
+
+        auto z = ComplexT{ZERO<TestType>()};
+        auto i = ComplexT{INVSQRT2<TestType>()};
+
+        SECTION("Apply using dispatcher") {
+            SECTION("CSWAP [0,1,2]|+10> -> |010> + |101>") {
+                const std::vector<ComplexT> expected_results = {z, z, i, z,
+                                                                z, i, z, z};
+
+                StateVectorKokkos<TestType> svdat012{num_qubits};
+                Kokkos::deep_copy(svdat012.getView(), ini_sv);
+
+                svdat012.applyOperation("CSWAP", {0, 1, 2}, false);
+
+                auto sv012 = Kokkos::create_mirror_view_and_copy(
+                    Kokkos::HostSpace{}, svdat012.getView());
+
+                for (std::size_t j = 0; j < exp2(num_qubits); j++) {
+                    CHECK(imag(expected_results[j]) == Approx(imag(sv012[j])));
+                    CHECK(real(expected_results[j]) == Approx(real(sv012[j])));
```

Comment:
Thanks for the suggestion! Because `sv012[j]` is a `kokkos::complex` this doesn't seem to work, and I have tried various ways to cast `sv012[j]` into a `std::complex` to make this work, but I can't find any good solutions. I will keep this for now

Reply:
Makes sense. Thank you for trying.